src/analysis/monotone-analyzer.h

```diff
@@ -0,0 +1,75 @@
+#ifndef wasm_analysis_monotone_analyzer_h
+#define wasm_analysis_monotone_analyzer_h
+
+#include <iostream>
+#include <queue>
+#include <vector>
+
+#include "cfg.h"
+#include "lattice.h"
+#include "wasm-traversal.h"
+
+namespace wasm::analysis {
+
+template<size_t N> struct MonotoneCFGAnalyzer;
+
+// A node which contains all the lattice states for a given CFG node.
+template<size_t N>
+struct BlockState : public UnifiedExpressionVisitor<BlockState<N>> {
```

Comment:
`UnifiedExpressionVisitor` is only needed if you implement `visitExpression` which I don't see here.

Reply:
Changed to visitor.